src/main/java/in/ac/bits/protocolanalyzer/mvc/controller/TestController.java

```diff
@@ -38,33 +37,48 @@
 	@Qualifier("Experiment")
 	private Experiment exp1;
 
-//	@RequestMapping(method = RequestMethod.GET)
-	public void runSequentialExp()throws Exception {
-		String pcapPath = "/home/vikram/Darshini/Test_pcap_files/sample_capture_test1.pcap";
-		String protocolGraphStr = "graph start {\n\tethernet;\n}\ngraph ethernet {\n\tswitch(ethertype) {\n\t\tcase 0800:			 ipv4;\n\t}\n}\ngraph ipv4 {\n\tswitch(protocol) {\n\t\tcase 06: tcp;\n\t}\n}\ngraph tcp {\n}\ngraph end {\n}";
-
-		exp1.init(pcapPath, protocolGraphStr);
-		exp1.analyze();
-		/*
-		TimeUnit.SECONDS.sleep(10);
-		log.info("] LAUNCHING EXPERIMENT TWO");
-		exp2.init(pcapPath, protocolGraphStr);
-		exp2.call();*/
-	}
-
-//	@RequestMapping(value = "/experiment", method = RequestMethod.GET)
-  @RequestMapping(method = RequestMethod.GET)
-	public @ResponseBody String runExp(@RequestParam("pcapPath") String pcapPath)throws Exception {
-		//String pcapPath = "/home/vikram/Darshini/Test_pcap_files/sample_capture_test1.pcap";
-		String protocolGraphStr = "graph start {\n\tethernet;\n}\ngraph ethernet {\n\tswitch(ethertype) {\n\t\tcase 0800:			 ipv4;\n\t}\n}\ngraph ipv4 {\n\tswitch(protocol) {\n\t\tcase 06: tcp;\n\t}\n}\ngraph tcp {\n}\ngraph end {\n}";
-
-		exp1.init(pcapPath, protocolGraphStr);
-		return exp1.analyze();
-		/*
-		TimeUnit.SECONDS.sleep(10);
-		log.info("] LAUNCHING EXPERIMENT TWO");
-		exp2.init(pcapPath, protocolGraphStr);
-		exp2.call();*/
+	@RequestMapping(method = RequestMethod.GET)
+	public @ResponseBody String runExp(@RequestParam("protocolGraphPath") String protocolGraphPath, @RequestParam("pcapPath") String pcapPath)throws IOException {
+		
+		String protocolGraphStr = null;
+		//Exception handlers for missing file / access issues / empty file for p4 graph
+		try {
```

Comment:
I know I gave this suggestion of putting try...catch here, but the logic is polluting the controller. It is better to have one / two private methods in Experiment class that perform checks on pcapPath and protocolGraphPath. They can give boolean answers. We can have an overloaded Experiment.init(pcapPath, protocolGraphPath) method that receives paths and invokes the private functions; the return values of private functions decide the return values of Experiment.init(pcapPath, protocolGraphPath) function. The existing Experiment.init(pcapPath, protocolGraphStr) can be invoked from with in the Experiment.init(pcapPath, protocolGraphPath) after verifying the protocolGraphPath value.

In a way
```java
int init(pcapPath, protocolGraphPath)  
{
    private protocolGraphStr;
    if (checkFileAccess(protocolGraphPath)) {
           protocolGraphStr = //read file into this string; if file length is zero, please throw FailedAnalysis() exception.
           init(pcapPath, protocolGraphStr)  
    } else
          throw new FailedAnalysis();
}    
int init(pcapPath, protocolGraphStr)  
{
    if (checkFileAccess(pcapPath)) {
           this.pcapPath = pcapPath;
	   this.protocolGraphStr = protocolGraphStr;
           init(pcapPath, protocolGraphStr)  
    } else
          throw new FailedAnalysis();
}    

```

In controller, the existing try...catch can be retained.
```java
		try {
 +			exp1.init(pcapPath, protocolGraphStr);
 +			experimentResults = exp1.analyze();
 +		}
 +		catch(Exception e) {
 +	    	return (new JSONObject()).put("packetCount", 0).put("status", "failure").put("comment", "Invalid Contents in file(s)").toString();
 +		}
```

Reply:
@prasadtalasila  while these tests check for validity of existence of files, they do not check for integrity of the contents.
Vague errors are shown as json output when we provide a non-empty file which is not a valid graph/pcap-file.